processor/src/lib.rs

```diff
@@ -161,6 +161,7 @@ where
     range: RangeChecker,
     chiplets: Chiplets,
     advice_provider: A,
+    max_cycles: Option<u32>,
```

Comment:
Now that I think about it, we could probably use `u32::MAX` as `max_cycles` when `max_cycles` is not specified. Could simplify things a bit here, and maybe also internally in `ExecutionOptions` we could store `max_cycles` as a simple `u32`.